ql/src/java/org/apache/hadoop/hive/ql/optimizer/stats/annotation/StatsRulesProcFactory.java

```diff
@@ -2921,6 +2920,97 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx procCtx,
+        return null;
+      }
+
+      final List<Operator<? extends OperatorDesc>> parents = lop.getParentOperators();
+      if (parents.size() != 2) {
+        LOG.warn("LateralViewJoinOperator should have just two parents but actually has "
+                + parents.size() + " parents.");
+        return null;
+      }
+
+      final Statistics selectStats = parents.get(LateralViewJoinOperator.SELECT_TAG).getStatistics();
+      final Statistics udtfStats = parents.get(LateralViewJoinOperator.UDTF_TAG).getStatistics();
+
+      final double factor = (double) udtfStats.getNumRows() / (double) selectStats.getNumRows();
+      final long selectDataSize = StatsUtils.safeMult(selectStats.getDataSize(), factor);
+      final long dataSize = StatsUtils.safeAdd(selectDataSize, udtfStats.getDataSize());
+      Statistics joinedStats = new Statistics(udtfStats.getNumRows(), dataSize, 0, 0);
+
+      if (satisfyPrecondition(selectStats) && satisfyPrecondition(udtfStats)) {
+        final Map<String, ExprNodeDesc> columnExprMap = lop.getColumnExprMap();
+        final RowSchema schema = lop.getSchema();
+
+        joinedStats.updateColumnStatsState(selectStats.getColumnStatsState());
+        final List<ColStatistics> selectColStats = StatsUtils
+                .getColStatisticsFromExprMap(conf, selectStats, columnExprMap, schema);
+        joinedStats.addToColumnStats(multiplyColStats(selectColStats, factor));
+
+        joinedStats.updateColumnStatsState(udtfStats.getColumnStatsState());
+        final List<ColStatistics> udtfColStats = StatsUtils
+                .getColStatisticsFromExprMap(conf, udtfStats, columnExprMap, schema);
+        joinedStats.addToColumnStats(udtfColStats);
+
+        joinedStats = applyRuntimeStats(aspCtx.getParseContext().getContext(), joinedStats, lop);
+        lop.setStatistics(joinedStats);
+
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("[0] STATS-" + lop.toString() + ": " + joinedStats.extendedToString());
+        }
+      } else {
+        joinedStats = applyRuntimeStats(aspCtx.getParseContext().getContext(), joinedStats, lop);
+        lop.setStatistics(joinedStats);
+
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("[1] STATS-" + lop.toString() + ": " + joinedStats.extendedToString());
+        }
+      }
+      return null;
+    }
+
+    private List<ColStatistics> multiplyColStats(List<ColStatistics> colStatistics, double factor) {
+      for (ColStatistics colStats : colStatistics) {
+        colStats.setNumFalses(StatsUtils.safeMult(colStats.getNumFalses(), factor));
+        colStats.setNumTrues(StatsUtils.safeMult(colStats.getNumTrues(), factor));
+        colStats.setNumNulls(StatsUtils.safeMult(colStats.getNumNulls(), factor));
+        // When factor > 1, the same records are duplicated and countDistinct never changes.
+        if (factor < 1.0) {
+          colStats.setCountDistint(StatsUtils.safeMult(colStats.getCountDistint(), factor));
```

Comment:
I think we should make sure that NDV is at least 1 in case numrows is >0

Reply:
Now I think this is available for this purpose if we add updating num trues and num falses?
https://github.com/apache/hive/blob/c082a724648b6bfbdd4b0ff72d7c41c29257beba/ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUtils.java#L2050-L2100

Reply:
This method may include additional logging and logics to optimize JOIN such as `cs.setFilterColumn`. It would be better to implement a simple and separate utility.

Reply:
Ceiled. I moved this method since I'd like to reuse it for HIVE-24240.
https://github.com/apache/hive/commit/50396346eaed5d6bab4ff87dd079918a769a7ebd